Added watched status and osd for playing from media library

zbyna opened the pull request:
Při přehrávání z media library přidává:
1. podporu pro automatické nastavení statusu shlédnuto  - při min 90 % přehrání položky
2. osd menu s posterem, dějem a obsazením  

Pro funkčnost náhledu v OSD je potřeba taky úprava stream resolveru: https://github.com/kodi-czsk/script.module.stream.resolver/pull/41

Testoval jsem dost, ale víte jako to chodí :smile: 

Pokud vyhovuje děkuji za merge.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kodi-czsk/plugin.video.sosac.ph/79)

<!-- Reviewable:end -->

zbyna commented:
První commit  je  s debug logy, druhý bez nich a ty zbývající jsou kočkovaná s codeclimate.
Sorry.

zbyna commented:
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

zbyna commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

hitman222 commented:
U mně se po úpravě přestali zobrazovat titulky u anglických filmů na sosáči,nebo jsem udělal někde chybu? Mám poslední LibreElec. Dík za odpověď.

zbyna commented:
Bez `debug logu` nejsem schopný k tomu říct ani ň 😄  Takže jenom dohady ...
U mě titulky v Jarvisu fungují a to jak po přehrání přímo z doplňku tak po přidání filmu či série do media library.
Měl jsem problem s  `service.LanguagePreferenceManager` https://github.com/ace20022/service.LanguagePreferenceManager/issues/3   - nevím jestli ji používáš - a zjistit by se to dalo zase z `debug logu`. Problém jsem snad odstranil zde: https://github.com/zbyna/service.LanguagePreferenceManager

hitman222 commented:
Kde ten debug log najdu, chybu mi to nevyhodí jen titulky nejdou(myslím titulky které jsou u ang filmu na sosáči automaticky),když titulky stáhnu tak ty jdou.

zbyna commented:
Debug log v LibreELEC  https://wiki.libreelec.tv/index.php?title=HOW_TO:Provide_Logfile#tab=Via_Kodi_Logfile_Uploader
Titulky moje změna neovlivňuje.
Spíš by mě zajímalo, jestli funguje na titulech přidaných do media library Osd při pozastavení přehrávání a jestli se nastaví příznak shlédnuto, když přehraješ alespoň 90 % titulu 😉 
To se snažím tímto pull requestem vyřešit.

hitman222 commented:
Ok,budu testovat ale blbne u nás internet tak nevím kdy.
 Jinak super práce díky moc.

hitman222 commented:
Tak u mně když video zastavím tak jde znovu od začátku a zhlédnuto taky neoznačí.

zbyna commented:
Tak to teda nevím ...  Testuju na win a takovéhle chování jsem nezaznamenal. Bez logu si tady můžem psát hodně dlouho, ale efekt to bude mít mizivý. Každopádně díky za test.

hitman222 commented:
Kam mužu poslat log?

zbyna commented:
Dej mu příponu txt a připoj ho do příspěvku tady na githubu.

hitman222 commented:
Místo log-2016-09-20-15.46.51.zip mám dát log-2016-09-20-15.46.51.txt to jsou kliky háky.

zbyna commented:
Rozpakovanej soubor, aby nebyly kliky háky, přejmenuješ na log.txt a připojíš ho do příspěvku.

zbyna commented:
Bylo by možné, aby někdo otestoval jak mu patch chodí ?
sosac:  https://github.com/zbyna/plugin.video.sosac.ph
stream.resolver:  https://github.com/zbyna/script.module.stream.resolver

Nastavení watched se mi po testování jeví jako hodně závislé na časování v : 

``` python
def play(self, item):
      ...
      while mujPlayer.isPlaying() == True:
                    xbmc.sleep(5000)
     ...
```

a 

``` python
def waitForChange(self):
        xbmc.sleep(4000)
        ...
```

Co chodí u mě nemusí chodit ostatním.
Dnes jsem časování opět prodloužil, ale nechci zatím dělat další pull requests.

Díky

hitman222 commented:
Tak jsem to otestoval na Windows a běží všechno OK,ale na LibreElec to neoznačí shlédnuto po min.90% ani se nezeptá kde pokračovat. Ještě jsem si všiml že Sosáč nenajde filmy jako Život je život nebo Než jsem tě poznala.

hitman222 commented:
[log-2016-09-21-08.36.44.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/484515/log-2016-09-21-08.36.44.zip)

hitman222 commented:
[log-2016-09-21-08.47.37.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/484520/log-2016-09-21-08.47.37.zip)

zbyna commented:
Díky za log !
Dobře že na win to chodí.
V linuxu to vypadá na problém s cestou k strm souboru, kterou tam nově ukládám ke klíči  title. 

``` Python
File "/storage/.kodi/addons/plugin.video.sosac.ph/resources/lib/sutils.py", line 75, in play
                                                pomItemType = JSON_result["result"]["filedetails"]["type"]
                                            KeyError: ('result',)
```

Buďto tam není vůbec nebo je ve špatném formátu a nic se pak nenajde:

``` Python
 +        if 'title' in item['info'].keys():
 +            pomTitle = xbmc.translatePath(item['info']['title'])
 +            JSON_req = {"jsonrpc": "2.0",
 +                        "method": "Files.GetFileDetails",
 +                        "params": {"file": pomTitle,
 +                                   "media": "video", },
 +                        "id": "1"}
 +            JSON_result = self.executeJSON(JSON_req)
 +            pomItemType = JSON_result["result"]["filedetails"]["type"].  
```

Přidal jsi filmy do media knihovny až po tom co jsi nainstalovat upravený sosač a stream resolver ?
Položky přidané před změnou přehrávat z media library nepůjdou.
Až se dostanu k počítači zkusim otestovat i v linuxu.

hitman222 commented:
Ano přidal jsem filmy až po instalaci nového sosáče a resolveru. 
Nevíš proč ten upravený sosáč nevidí filmy s diakritikou?

zbyna commented:
Netuším, do kódu pro hledání jsem nezasahoval, možná side effect.

iChUdY commented:
Čau, zkoušel jsem na iOS a nejde přehrávat z knihovny. Z doplňku přehrávat lze. 

Dnes se k tomu dostanu snad na dýl.

zbyna commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

hitman222 commented:
Tak po upravě zatím všechno po starém,nic se nezměnilo. Tady je log
[log-2016-09-22-16.24.03.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/487813/log-2016-09-22-16.24.03.zip)

hitman222 commented:
A v OpenElecu video z knihovny nejde přehrát.
[log-2016-09-22-16.27.01.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/487818/log-2016-09-22-16.27.01.zip)

zbyna commented:
@hitman222 

##### 1.  Log-2016-09-22-16.24.03.zip - LibreELEC release: RPi2.arm-7.0.2
- tady žádnou chybu sosače nemůžu najít, je tu pouze:

``` python
10:28:48 1022.313354 T:1796207520   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'urllib2.HTTPError'>
                                            Error Contents: HTTP Error 404: Not Found
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.uktvnow/default.py", line 1,
```
- když tak zkus addon plugin.video.uktvnow odinstalovat a vyzkoušet pak

##### 2.   Log-2016-09-22-16.27.01.zip  -  OpenELEC release: RPi2.arm-6.95.3
- tady je ta samá hláška jako minule:

``` python
File "/storage/.kodi/addons/script.module.stream.resolver-master/lib/contentprovider/xbmcprovider.py", line 71, in run
                                                return self.play({'url':params['play'],'info':params})
                                              File "/storage/.kodi/addons/plugin.video.sosac.ph-master/resources/lib/sutils.py", line 75, in play
                                                pomItemType = JSON_result["result"]["filedetails"]["type"]
                                            KeyError: ('result',)
                                            -->End of Python script error report<--
```

vypadá to, jako by tam ta poslední změna kódu nedorazila protože číslo řádku

``` python
 pomItemType = JSON_result["result"]["filedetails"]["type"]
```

se v upraveném kódu posunulo z 75 na 78  

 taky se u tebe změnl adresář pluginu (plugin.video.sosac.ph   vs  plugin.video.sosac.ph-master), ale to nemusí hrát roli
- já nemám malinu a linux testuju  na dualbootu windows /mint případně na virtuálu. Tam mi přehrávání chodí  v obou systémech.

hitman222 commented:
Tak uktvnow odinstalovaný a všechno je po starém, žádná změna, tady je log po dnešním testu.
[log-2016-09-23-06.45.52.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/489041/log-2016-09-23-06.45.52.zip)
Je to škoda na windows to vypadá a hraje suprovně.
Zkoušel jsem ještě změnit adresář knihovny z USB do paměti LibreElecu,ale nepomohlo.

iChUdY commented:
Ahoj, video z knihovny v LE8 na malině nejde přehrát:

11:16:58 163.112930 T:1634202528   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: ('id',)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sosac.ph-master/default.py", line 47, in <module>
                                                XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps), settings, **addon**).run(params)
                                              File "/storage/.kodi/addons/script.module.stream.resolver-master/lib/contentprovider/xbmcprovider.py", line 71, in run
                                                return self.play({'url':params['play'],'info':params})
                                              File "/storage/.kodi/addons/plugin.video.sosac.ph-master/resources/lib/sutils.py", line 79, in play
                                                pomItemDBID = JSON_result["result"]["filedetails"]["id"]
                                            KeyError: ('id',)
                                            -->End of Python script error report<--
11:16:58 163.257950 T:1961553920   ERROR: Playlist Player: skipping unplayable item: 12, path [plugin://plugin.video.sosac.ph/?play=687474703a2f2f74762e7072656872616a2e6d652f63732f706c617965722f73696d70736f6e6f76692d73312d653133&cp=736f7361632e7068&title=7370656369616c3a2f2f70726f66696c652f6164646f6e5f646174612f706c7567696e2e766964656f2e736f7361632e70682f545653686f77732f53696d70736f6e6f76692f536561736f6e20312f53314531332e7374726d&]

Přehrávat z doplňku jde...

zbyna commented:
@hitman222 
v logu žádnou chybu spojenou se sosačem nevidím, pořad se tam motá něco z toho plugin.video.uktvnow, je to jako zakletý
@iChUdY 
tady to vypadá že hledaný název strm souboru v media library nebyl nalezen, ale netvrdím že tam není 😄 

Mallinu nemám, na virtuálu - Linux Mint 17.3  a ve win 7 to běhá víceméně spolehlivě 😕 
Doplním do sosače testovací logy, protože takhle mám málo informací. Comitnu úpravy do
https://github.com/zbyna/plugin.video.sosac.ph a dám tady vědět. Tenhle víkend to ale s velkou pravděpodobností nebude.

zbyna commented:
Dneska jsem si vypůjčil od kamaráda malinu (rpi2):
- nainstaloval jsem libreElec image `LibreELEC-RPi2.arm-7.0.2.img.gz` pomocí http://releases.libreelec.tv/LibreELEC.USB-SD.Creator.Win32.exe  
- knihovnu sosače jsem nechal na defaultním umístění
- přidal film Ohněm a mečem do library

a přehrávání z media library mi chodí bez problémů

Problém je pouze s nastavením statusu shlédnuto a to v linuxu a vypadá to že linux narozdíl od win prostě nebere:

``` python
while mujPlayer.isPlaying() is False:
```

ale pouze

``` python
while mujPlayer.isPlaying() == 0:
```

což mě teda dost překvapilo :scream:  :smile: a to tak, že už nějak nemám sílu připravit pull request :sleepy:
asi bude lepší psát:

``` python
while not mujPlayer.isPlaying() :
```

Jinak jsem připravil zjednodušené reportování chyb pomocí modulu bugallo (defaultně nainstalovaný v Kodi) :  https://github.com/zbyna/script.module.buggalo
kdy v případě chyby na uživatele vykoukne okno s odesílanými informacemi a ten jedním kliknutím odešle vývojářem předem definované informace do mailu a nemusí  řešit logy . Ale to už asi nebude potřeba.

hitman222 commented:
while not mujPlayer.isPlaying() : co to pro nás znamená když máme malinu?

zbyna commented:
Buďto bude v kódu fungovat

``` python
while not mujPlayer.isPlaying():
```

pro win i linux (taky malina) nebo se po testu na platformu vybere kód ,který tam funguje.
Doplním to  do pull requestu, ale teď to zrovna časově nebude moc vycházet no.

Když to shrnu tak:
- přehrávání mi chodí ve stávající verzi pull requestu na win, virtuálu linux-mint 17.3 a po čisté instalaci zapůjčené maliny i tam (viz. minulý post)
- nastavení statusu shlédnuto chodí pouze ve win a řešení  viz. začátek tohoto postu

iChUdY commented:
Ahoj,

čistá instalace Jarvis na OSX. Nepřehrává z knihovny:

15:01:21 T:123145319092224   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: ('id',)
                                            Traceback (most recent call last):
                                              File "/Users/chudy/Library/Application Support/Kodi/addons/plugin.video.sosac.ph-master/default.py", line 47, in <module>
                                                XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps), settings, **addon**).run(params)
                                              File "/Users/chudy/Library/Application Support/Kodi/addons/script.module.stream.resolver/lib/contentprovider/xbmcprovider.py", line 71, in run
                                                return self.play({'url':params['play'],'info':params})
                                              File "/Users/chudy/Library/Application Support/Kodi/addons/plugin.video.sosac.ph-master/resources/lib/sutils.py", line 79, in play
                                                pomItemDBID = JSON_result["result"]["filedetails"]["id"]
                                            KeyError: ('id',)
                                            -->End of Python script error report<--
15:01:21 T:140735323398144   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.sosac.ph/?play=687474703a2f2f6d6f766965732e7072656872616a2e6d652f63732f706c617965722f7465726d696e61746f722d322d64656e2d7a7563746f76616e692d31393931&cp=736f7361632e7068&title=7370656369616c3a2f2f70726f66696c652f6164646f6e5f646174612f706c7567696e2e766964656f2e736f7361632e70682f4d6f766965732f5465726d696e61746f7220322044656e207a7563746f76616e69202831393931292f5465726d696e61746f7220322044656e207a7563746f76616e69202831393931292e7374726d&]

čistá istalace Jarvis na iOS9 z knihovny přehrává, ale poster a název v OSD pořád při starým. Watched status taky nic.

hitman222 commented:
OK, dík za objasnění. Upravit to musíš celé ty, nebo stačí upravit jen ten jeden řádek a zvládnu to sám?

iChUdY commented:
Ahoj,

čistá instalace Jarvis na Windows 10:

-watched status nefunguje(koukal jsem poctivě na celý 20min díl)
-v OSD je u filmu pouze název a poster chybí, u seriálu pouze název dílu(zde by se hodil i název seriálu), poster chybí
-přehrávat z knihovny jde :+1:

zbyna commented:
@hitman222 
Opravím to, až budu mít čas, pokud máš dost času na experimenty, zkus to - nemusí to být o úpravě jednoho řádku, můžou se objevit další komplikace ...

@iChUdY 
Díky za logy.
Mac
- to je stejná chyba jakou jsi měl na malině (pokud si to dobře pamatuju) tj. soubor nebyl v databází nalezen, tohle nedokážu vysvětlit a klonil bych se spíš k tomu, že máš nějakou speciální konfiguraci třeba sdílení media library mezi instalacemi ? Prostě nevim. Příště postni celý log, může tam být nějaké další vodítko.

Win 10
- tady tě zklamu win 10 mi do baráku nesmí 😄  A to ani na virtuál. Pokud mám mít šanci to vyřešit potřebuju celý log.

zbyna commented:
_[a discussion](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79#-KSnSTZA-wFpG8IXRN6F:-KSnSTZA-wFpG8IXRN6G:btmrkd0) (no related file):_
Opraveno nastavení statusu shlédnuto v linuxu.
Podle mých testů funguje při přehrávání z media library (Jarvis 16.1) :

| feature | win 7 | linux | osx |
| --- | --- | --- | --- |
| OSD  (náhled + děj + obsazení ) | ano | ano | ano |
| nastavení statusu shlédnuto (min 90 % přehrání ) | ano | ano | ano |

linux -  mint 17.3 virtuálně vmware, rpi2 LibreElec 7.0.2 - nativně půjčená :smile: 
win 7, osx (el capitan) - nativně

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

zbyna commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

hitman222 commented:
Tak to je super,kdy můžeme testovat?

ExSport commented:
@hitman222 Hned: https://github.com/zbyna/plugin.video.sosac.ph
Clone or Download --> Download ZIP --> Install

hitman222 commented:
Ok, díky.

pstruh commented:
U mna nejde nic prehrat z TV Shows library. Priamo z doplnku ide v pohode. Errors v logu.
http://sprunge.us/gCjE

RPi2 - LibreELEC (Milhouse) - Version: devel-20160831210440-#0831-gec9cf07 [Build #0831] 8.0

zbyna commented:
@pstruh 
Díky za log.  
Provozuješ beta verzi Kodi 17  tzn. z povahy věci neodladěnou a nestabilní.
Tady je varování z http://forum.kodi.tv/showthread.php?tid=269815:

```
Fair Warning
Kodi 17 is currently beta and there is likely to be a lot of activity in the coming weeks/months. 
There will be some breakage.
In addition, these are "bleeding edge" builds - they often contain cherry-picked commits that
require early testing prior to merging upstream, and which won't be found in regular nightlies
until after they have been merged.
```

Nezlob se, ale dokud bude  LibreElec a Kodi 17 v něm **beta** nebudu pro ně změny upravovat.
Mám jenom jedny nervy :smile: 
V postu s tabulkou jsou verze a systémy na kterých jsem úpravy otestoval a fungovaly, takže budou s
velkou pravděpodobností fungovat i na jiných konfiguracích.

zbyna commented:
Opraveno:

``` python
KeyError: ('result',)
KeyError: ('id',)
```

na všech platformách - doufám :smile: 

Cesta k souboru se v media library ukladá klasicky:  `directory1/directory/` nebo pomocí specialního KODI protokolu např. `special://profile`  a to nezávisle na platformě - i když jsem si původně myslel opak - jo, jo - chybama se člověk učí :sweat_smile: 

Snažil jsem se zjednodušit testování (a taky sobě ladění :smile:  ) pomocí addonu Buggalo Exception Collector z oficiálního repozitáře Kodi (http://kodi.wiki/view/Add-on:Buggalo_Exception_Collector).

Na vybraných místech to při chybě vyhodí okýnko s informacemi, které se v případě uživatelova souhlasu pošlou na web. Ten jsem připravil tady:  http://sosac.comli.com/index.php  a je z něho vidět na jakých verzích a systémech úprava funguje. Detaily chyby jsou přístupné jenom autorizovaným uživatelům, pokud by někdo chtěl web použít při  ladění vlastních úprav, přístup zřídím.

Podpora Buggalo Exception Collector vyžaduje v sosači o závislost víc, pokud to repo maintainers vidí jako problém, závislost odstraním.

Děkuji za případný test

zbyna commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

hitman222 commented:
Odzkoušeno už jsem ti poslal 2 reporty,používám upravenou verzi 1.2.8 kde se zobrazují filmy s diakritikou,ale zase chybí lepší obrázky.

zbyna commented:
@hitman222 Díky. 
Pro testování by bylo lepší updatovat sosač na poslední změny z: 
https://github.com/zbyna/plugin.video.sosac.ph
K chybě:

``` Python
type 'exceptions.ValueError' time data '' does not match format '%H:%M:%S'
Stacktrace  
```

dochází vždy, nebo jenom při druhém a dalším pokusu o přehrání od restartu maliny ?

Ve vzhled / nastavení / formát času máš h:mm:ss a pod tím 24 hodinový formát ?

Mi se tuto chybu nedaří zopakovat a to mám na malině (položka 6 na sosac.comli.com) stejnou verzi kodi, lišíme se "pouze " jádrem: Linux 4.4.13 armv7l vs Linux 4.1.19v7-aufs armv7.

Na jakém filmu testuješ ?

pstruh commented:
Mozem potvrdit v Kodi 17 beta funguje posledna uprava skleve :) zatial som nenarazil na ziadny problem, da prehravat z library aj oznacuje watched status po skonceni, takze spokojnost.
VDAKA!
RPi2 - LibreELEC (Milhouse) - Version: devel-20160831210440-#0831-gec9cf07 [Build #0831] 8.0

zbyna commented:
@pstruh 
Díky, dobře se to čte :smile:

pstruh commented:
Tak predsa len jeden problem, sosac sa nevypne a zamrze obrazovka. Uz sa tu ten problem riesil tu: https://github.com/kodi-czsk/plugin.video.sosac.ph/issues/41

CPythonInvoker(12, /storage/.kodi/addons/plugin.video.sosac.ph-master/default.py): script didn't stop in 5 seconds - let's kill it

http://sprunge.us/WOSV

zbyna commented:
@pstruh 
Nejde vypnout pokaždé nebo jenom v určitých situacích ? Na malině se mi tu chybu nedaří zopakovat.  Naslepo jsem použil řešení z https://github.com/kodi-czsk/plugin.video.sosac.ph/issues/41 viz. patch df19467 .
Teď mě ještě došlo, že pokud se pustí z media library odkaz (strm soubor), který nebyl vytvořený touto verzí sosače, tak taky dojde právě k tomu co popisuješ. Nevím jestli má smysl vytvářet nástroj na migraci strm souborů na nový formát...  Ale teď už jdu do hajan :smile:

pstruh commented:
Z library ked dam otvorit napr. nejaky odkaz kde neexistuje este uploadnute video a ked dam exit tak to zamrzne. Robi to aj ked mi ponukne prehrat kvalitu stremu ale nasledne to zrusim.
Skusal som len existujuce polozky v library. Vyskusam este nanovo pridat do library.

zbyna commented:
@pstruh 
Zamrzání by mělo být vyřešeno:
- při zrušení výběru kvality určitě
- neuploadnuté video snad :sweat_smile:  

Starý formát strm souborů zatím neřeším.

pstruh commented:
Zdravim,
ano funguje to, Ukonci sa to korektne aj z Library aj priamo z doplnku. VDAKA :)

zbyna commented:
Reviewed 2 of 2 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/kodi-czsk/plugin.video.sosac.ph/79)_

<!-- Sent from Reviewable.io -->

zbyna commented:
Opraveno:

``` python
type 'exceptions.ValueError' time data '' does not match format '%H:%M:%S'
```

doufám na všech platformách :smile: 

Přidána možnost převést  .strm soubory vytvořené v minulých verzích sosače na nový
formát, který umožňuje při přehrávání z media library spolupráci s OSD a automatické nastavení  `shlédnuto`
Před testováním migrace formátu  si radši udělejte zálohu strm souborů.

Díky za test.

hitman222 commented:
U serialu vyhazuje chybu sosace,tady je log. 
[log-2016-10-15-06.40.44.zip](https://github.com/kodi-czsk/plugin.video.sosac.ph/files/530801/log-2016-10-15-06.40.44.zip)

zbyna commented:
@hitman222 
Jestli máš na mysli tuto:

``` Python
ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--   - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: cannot concatenate 'str' and 'list' objects
                                            Traceback (most recent call last):

File "/storage/.kodi/addons/plugin.video.sosac.ph/default.py", line 47, in <module>
XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps), settings, __addon__).run(params) 
File "/storage/.kodi/addons/script.module.stream.resolver/lib/contentprovider/xbmcprovider.py"
, line 66, in run  
                            self.list(self.provider.list(params['list']))
 File "/storage/.kodi/addons/plugin.video.sosac.ph/resources/lib/sosac.py", 
line 195, in list 
                             return self.list_tv_shows_by_letter(url)
File "/storage/.kodi/addons/plugin.video.sosac.ph/resources/lib/sosac.py", line 528, 
in list_tv_shows_by_letter
                           util.info("Resloved shows " + shows)
                          TypeError: cannot concatenate 'str' and 'list' objects
 -->End of Python script error report<-
```

tak ta tam spadla zřejmě z ninulèho merge  https://github.com/kodi-czsk/plugin.video.sosac.ph/pull/84/commits/c528fd2b692d5419737dae6368b1fce679d53d57#diff-aa40f6b20de55e538cf446c7434aab92L528  , který jsem přebral z hlavního repozitáře sosače.

@ExSport 
Třeba se pletu, ale zřejmě je něco s tou úpravou logování  ?
https://github.com/kodi-czsk/plugin.video.sosac.ph/pull/84

ExSport commented:
@zbyna
Dik. Já věděl, že při tom copy-paste mě něco proklouzne :+1:

ExSport commented:
@zbyna Opraveno v https://github.com/kodi-czsk/plugin.video.sosac.ph/pull/87

zbyna commented:
@ExSport 
To je teda rychlost 👍  a v sobotu večer.
Díky.

@hitman222 
Až to commitmou převezmu úpravu do svého forku.

hitman222 commented:
Ok, díky za super práci... 👍😉

zbyna commented:
Pro toho kdo postnul `issue id 12` na http://sosac.comli.com

``` Python
Type    <type 'exceptions.ValueError'>
Message type 'exceptions.ValueError' time data '' does not match format '%H:%M:%S'
Stacktrace
File "/storage/.kodi/addons/plugin.video.sosac.ph-master/resources/lib/myPlayer.py", 
line 79, in onPlayBackStarted

    xbmc.getInfoLabel('Player.TimeRemaining(hh:mm:ss)'))

  File "/storage/.kodi/addons/plugin.video.sosac.ph-master/resources/lib/myPlayer.py", 
line 53, in get_sec

    t = datetime.strptime(time_str, "%H:%M:%S")

  File "/home/chewitt/LibreELEC.70-official/build.LibreELEC-RPi2.arm-7.0.2/Python-2.7.11/.install_pkg/usr/lib/python2.7/_strptime.py", line 325, in _strptime

Execution
Python  2.7.11 (default, Jun 22 2016, 17:52:25) 
```

Nepoužíváš poslední verzi změn z:  https://github.com/zbyna/plugin.video.sosac.ph
Chyba byla opravena:
https://github.com/zbyna/plugin.video.sosac.ph/commit/5fe08f9429f70d15636e9d57c6e1ad06fa7b1916?diff=split#diff-117db81edc55f4af273f8daa344202c8L79

zbyna commented:
@hitman222
Nakonec jsem přidal úpravu od ExSporta sám + update posledního commitu z master sosače + aktualizace knihovny po nastavení shlédnuto

hitman222 commented:
Super, hned to jdu nahodit... 👍

pstruh commented:
Pri nacitani filmov podla pismena zobrazuje vestko mozne aj filmy mimo tej skupiny. Napr. pod filmamy A vidim aj filmy pod inym menom.
Download nefunguje, nesiel sice aj predtym len som si chcel nieco stiehnut a hned error. Neviem ci to nebolo umyselne znemoznenie downloadu, mne moc nechyba :)
http://sprunge.us/Pdjh

este som si vsimol ze pri niektorych serialoch mam subory v kniznici dva krat. Ked to vymazem a pridam nanovo je to tak jeden krat, zapnem na druhy den a zase ich mam dva krat. Asi nieco prehliadam?

ExSport commented:
@pstruh Kde jsi vzal, že download nefungoval?
Od verze v1.2.6 by měl fungovat, viz můj fix: https://github.com/kodi-czsk/plugin.video.sosac.ph/commit/c633d8361ed7841e5c2cf056be6ec1b54128a823#diff-aa40f6b20de55e538cf446c7434aab92R513

zbyna commented:
@pstruh  @ExSport 
Chyba:

``` Python
-->Python callback/script returned the following error<--- 
NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                      Error Type: <type 'exceptions.TypeError'>
                       Error Contents: save_to_file() takes exactly 2 arguments (3 given)
 Traceback (most recent call last):
File
 "/storage/.kodi/addons/plugin.video.sosac.ph-master/default.py",
 line 47, in <module>
    XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps), settings, __addon__).run(params)  
File
 "/storage/.kodi/addons/script.module.stream.resolver/lib/contentprovider/xbmcprovider.py", 
line 69, in run
         return self.download({'url':params['down'],'title':params['title']})
 File 
"/storage/.kodi/addons/script.module.stream.resolver/lib/contentprovider/xbmcprovider.py", 
line 158, in download
    util.save_to_file(stream['subs'],
                                os.path.join(downloads,name+'.srt'),
                                stream['headers'])
     TypeError: save_to_file() takes exactly 2 arguments (3 given)
-->End of Python script error report<--
```

je možná :smile: způsobená záměnou 
funkce `save_to_file z util.py`

``` Python
def save_to_file(url, file):
    try:
        return save_data_to_file(request(url), file)
    except:
        traceback.print_exc()
```

za: `save_to_file z xbmcutil.py`

``` Python
def save_to_file(url, file, headers=None):
    try:
        f = open(compat_path(file), 'w')
        f.write(request(url, headers))
        f.close()
        return True
    except:
        traceback.print_exc()
```

Mýma úprava by to být nemělo, tak na 90 % :smile:

Vůbec nepochybuju, že to v 1.2.6 bylo fixnuto.
Pravděpodobně to souvisí s https://github.com/kodi-czsk/script.module.stream.resolver/commit/d84a721372660f8dc63f812174fbcbbc42291fc4

@pstruh 
K tomu zdvojení potřebuju víc informací:
Dělal jsi mezitím migraci strm souborů na nový formát ?
Zdvojí se pouze záznam v knihovně nebo i strm soubory ?

pstruh commented:
Tak to zdvojenie som vyriesil, bola to moja chyba. Tie serialy pri ktorych som mal zdvojenie som zabudol vymazat z predchadzajuceho adresara kde som mal nastavenu cestu na ukladanie strm suborov. Teraz pouzivam default nastavenia Sosaca, len neviem preco mi tahalo strm subory aj z predchadzajucej lokacie.

ExSport commented:
Tak poslední verze Sosace i Resolveru mě funguje taky.
Problém je, že ani po zaslaném download fixu jsem netestoval video s titulky, takže je možné, že nefungoval download s titulky ani předtím.
Teď jsem zkusil poslední verzi Sosace+Resolveru a opravdu video s **titulky** nelze stáhnout.

zbyna commented:
@pstruh 
:+1:  
 možná máš ten předcházející adresář jako zdroj pro knihovnu medií, jinak nevim

@ExSport 
Teďka co opravit ?  :smile: 
`save_to_file z util.py` na:

``` python
def save_to_file(url, file, headers = None):
    try:
        return save_data_to_file(request(url, headers), file)
    except:
        traceback.print_exc()
```

a zřejmě další závislosti  v util.py

nebo 

``` python
v "/storage/.kodi/addons/script.module.stream.resolver/lib/contentprovider/xbmcprovider.py", 
line 158, in download
   util.save_to_file(stream['subs'],
                          os.path.join(downloads,name+'.srt'),
                          stream['headers'])
```

na:

``` python
 xbmcutil.save_to_file(stream['subs'],
                        os.path.join(downloads,name+'.srt'),
                        stream['headers'])
```

zbyna commented:
Přidal jsem:
- progress bar  pro hledání a změnu strm souborů při migraci starého formátu a
- přeložil
  použité stringy do en,cs a slovenštiny - tady se omlouvám za případné chyby protože slovensky rozumím, ale další drobné niance mi veskrze unikají :smiley: 

Nebudu už nic dalšího vymýšlet a pokud se neobjeví žádné další problémy a chyby, squashnu všechny
commity do jednoho a budu čekat na případný merge do kodi-czsk:master .

zbyna committed:
Added watched status and osd for playing from media library

for proper functioning  merging https://github.com/kodi-czsk/script.module.stream.resolver/pull/41
needed

zbyna commented:
Případné konflikty budu řešit až po commitu:
https://github.com/kodi-czsk/plugin.video.sosac.ph/pull/85 https://github.com/kodi-czsk/plugin.video.sosac.ph/pull/87
protože jejich část jsem kvůli funkčnosti dolpňku ve svém forku
(možná to byla chyba) už zohlednil.
Pokud vadí něco dalšího děkuji za upozornění.

lubo commented:
Preco radsej nepridad tuto feature do `xbmcprovider` kniznice, aby ju mohli lahko zacat pouzivat aj ostatne pluginy ?

Dalsia vec je, ze si do jedneho PR napchal nesuvisiace veci, ako napr. podpora pre Buggalo.

zbyna commented:
@Lubo 
Tvůj způsob vyjadřování (napchal ?!)  se mi vůbec nelíbí.   Hraješ si na Linuse ?  Do něj máš daleko :smiley:  zrovna tak jako já :smiley: 

Buggalo podle mého názoru s touto úpravou souvisí, protože pokud jsi si nevšimnul, slouži k účinnějšímu ladění a podstatně zjednodušuje zapojení i né tak pokročilých uživatelů, kterí mi mimochodem dost pomohli v odchytávání chyb. Jak jsem psal, klidně jeho podporu zruším.
Mohl bys mi prosím výjmenovat další, podle tebe, nesouvisející věci ? Děkuji.

`xbmcprovider` rozšiřovat nemíním, určitě existují další cesty jak `watched` implementovat a zrovna jsem se dostal do časového presu, takže sosáč musí stranou.

lubo commented:
@zbyna, na nic sa nehram. Opisujem veci tak ako su. Za to ty sa na cloveka, ktory si obcas nevie dat 2 a 2 dokopy, celkom dost nafukujes. (Mala rada do zivota, nehodnot ludi, ktorych nepoznas, alebo sa prv aspon uisti, ze si lepsi nez oni ;) )

Ja dobre viem, co je Buggalo, je mi jasne, ze ti dost pomohlo a ze by aj nam pomohlo, avsak nie, nesuvisi s touto samotnou feature. Na druhej strane z toho nemusis byt hned na nervy. Ked sa ukludnis, tak mozes na to spravit samostatny PR a tam sa o tom pobavime viacej.
Hlbkovo som zatial nic nereviewoval, ale bud si isty, ze ak nieco najdem, tak ti dam vediet.

Ok. Ja ale potrebujem dobre argumenty na to, aby som cokolvek mergol a preto teda potrebujem argumenty na to, preco by toto malo ist iba sem nie do `xbmcprovider`.

zbyna commented:
@Lubo 
Tam, kde slušný člověk použije kouzelná slova jako promiň, omlouvám se, moc si to bereš apod, ten neslušný začne dávat dobré rady, osočovat z nafukování, chovat se povýšeně. Tohle není o tom, kdo
je lepši, ale o tom kdo chce být slušný.
Takže NE nehodnotím tebe ani tvoje vědomosti. ANO nelíbí se mi způsob jakým se mnou komunikuješ.

lubo commented:
@zbyna Prepac, ale tvoje pocity, nazory, domienky a idei nikoho vonkoncom nezaujimaju, iba ze by boli sucastou toho PR. Ja som iba polozil prostu otazku a doplnil jednoduchu pripomienku. Ak nevies, tak review nie je zdrbanie tvojej prace, ale je to diskusia, v kt. mas obhajit svoju pracu. No a teraz si tento PR zavrel bez udania akehokolvek dovodu. Co si o tom mame mysliet ? Chyba tomu nejaky zaver.

hitman222 commented:
Čau, tak co je nového?

iChUdY commented:
Coby. Zbyna na todle nemá náladu a bude to dělat jen pod sebou. Na potřebné změny v sosáči se čekalo roky a když konečně přijde někdo kdo to napíše, tak nic... Pro mě je sosáč už dávno mrtvý už jen proto že se to dávno nejmenuje sosáč viz. https://github.com/iChUdY/plugin.video.prehraj.me

lzoubek commented:
To jsme si mohli psat vsichni pluginy jen pro sebe, je to mnohem jednodussi - nemusis s nikym diskutovat co a jak udelat. 

Tyhle review slouzi prave proto, aby se do pluginu nedostal polofunkcni kod - a jakoze se sem uz dostal - cely kod, ktery resi pridavani do knihovny jsem nedostatecne prohledl a otestoval a proto ma pridavani do knihovny tolik bugu.

@iChUdY Ohledne zivota sosace .. sosac mrtvy urcite neni. Nazev pluginu na tom nic nemeni - je to na pluginu asi ta nejmene podstatna vec. To, ze jsi ho forknul a udelal z nej neco jineho a mozna lepsiho je dobre. Je to ale dobre hlavne pro tebe a par zasvetsenych.

lubo commented:
Pozrite sa, toto je community-driven projekt a prave komunita udava, ze ci je tento projekt mrtvy alebo nie. Dalsia vec je, ze kazdy clen komunity musi respektovat sukromny cas toho druheho, cize aj maintainerov. Dalej, konkretne tento PR som neriesil od uplneho zaciatku, pretoze prave komunita si sama spravila peer review a tak to ma byt. Za tie 2 mesiace tu bezny clenovia diskutovali s danym koderom, pisielali feedback a cele sa to niekam hybalo. Teraz som si vsimol, ze sa to uz dostalo do nejakeho rozumneho bodu a zacal som svoj review, ktory je robeny so vsetkym respektom k tomuto projektu a praci kazdeho z nas, ktoru tu odviedol. Ano, moja komunikacia je velmi priama a stroha, lebo tak je to optimalne. Preto prosim kazdeho jedneho aby sa zamyslel pred tym, nez sem cokolvek postne, ze co od toho vlastne ocakava. Ja osobne totiz ocakavam nejake vysledky, nie nejake bratrickovanie sa alebo podobne.

A ak niekto napise nieco typu "Zbyna na todle nemá náladu...", tak by som bol rad, keby si dotycny laskavo uvedomil, ze:
1. Zbyna bol ten, kto sem chcel dostat svoju pracu (za co sme samozrejme vdacni)
2. Zbyna bol ten, kto zacal riesit svoje pocity a dojmi miesto absolutne korektnych otazok, ktore boli kladene na margo jeho prace
3. Zbyna bol ten, kto psychicky nevydrzal ten neskutocne maly tlak, ktory bol vyvijany na kvalitu jeho prace (ako @lzoubek napisal, musime si skratka udrzovat nejake standardy kvality, inac tym utrpia vsetci)

Samozrejme, nehladiac na to, ze ci sa zbyna dalej planuje zapajat alebo nie, nemusi to tu skoncit. Kludne mozete niekto dalsi forknut jeho repo, poslat novy PR a budeme pokracovat v review tam, kde to skoncilo.

hitman222 commented:
Nevím co jste si podělali, ale @zbyna  odvedl výbornou práci všechno jede jak má za co mu dávám velké DÍK.
Je pravda,že by jsme se tady nemuseli mezi sebou hádat a spojit všechno dohromady v prospěch výborného pluginu.

lubo commented:
"Nevím co jste si podělali, ..." - to, co tu predvadzate, nie je pro-komunitny pristup.
